### PR TITLE
Promote DNS e2e verifying DNS resolution for externalName services

### LINKED ISSUE #73425: Tracking Issue - Conformance Coverage for Networking
**What would you like to be added**:
This is a place holder to track Conformance Coverage on the Topic.
This issue should remain open until all related work is accomplished in the k/k repo
This issue will contain analysis of the coverage. Existing tests and additional tests requested with links to those issues/PR's

**Why is this needed**:
Specifically address CNI networking, identify tests that tests pod-pod communication

/area conformance

### PR DESCRIPTION
**Pull request by mgdevstack**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Promotes DNS e2e which is verifying DNS resolution behavior for externalName services and clusterIP services.

**Which issue(s) this PR fixes**:

Fixes #73425

**Special notes for your reviewer**:
- Maybe failing for windows.
- Not Flaking
- Promoting to Conformance as it verifies important functional scenario of resolving DNS for services pointing to the external domain and for service of type clusterIP.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
/area conformance
/sig testing
@kubernetes/sig-network-pr-reviews

### CODE REVIEW

**Review by johnbelamaric — CHANGES_REQUESTED** [review]

File: test/e2e/network/dns.go [hunk 1]
```diff
@@ -247,7 +247,13 @@ var _ = SIGDescribe("DNS", func() {
 		validateDNSResults(f, pod1, append(wheezyFileNames, jessieFileNames...))
 	})
 
-	It("should provide DNS for ExternalName services", func() {
+	/*
+		Release: v1.14
+		Testname: DNS, for ExternamName Services
+		Description: Create an service with externalName. Pod MUST be able to resolve DNS for this service by serving CNAME. When externalName of this service is changed, Pod MUST resolve to new DNS entry for the service.
```
[the remark this answers is not in the document]
Comment (mgdevstack) [reply test/e2e/network/dns.go]: Thanks for reviewing. Included suggested changes.

File: test/e2e/network/dns.go [hunk 2]
```diff
@@ -247,7 +247,13 @@ var _ = SIGDescribe("DNS", func() {
 		validateDNSResults(f, pod1, append(wheezyFileNames, jessieFileNames...))
 	})
 
-	It("should provide DNS for ExternalName services", func() {
+	/*
+		Release: v1.14
+		Testname: DNS, for ExternamName Services
```
[the remark this answers is not in the document]
Comment (mgdevstack) [reply test/e2e/network/dns.go]: Done.

File: test/e2e/network/dns.go [hunk 3]
```diff
@@ -247,7 +247,13 @@ var _ = SIGDescribe("DNS", func() {
 		validateDNSResults(f, pod1, append(wheezyFileNames, jessieFileNames...))
 	})
 
-	It("should provide DNS for ExternalName services", func() {
+	/*
+		Release: v1.14
```
[the remark this answers is not in the document]
Comment (mgdevstack) [reply test/e2e/network/dns.go]: Done.

**Comment by bclau** [comment]
This test is known to pass on Windows on Azure (azure-vnet-cni), GCE, vsphere, flannel overlay and l2bridge. LGTM.

/lgtm

File: test/e2e/network/dns.go [hunk 4]
```diff
@@ -255,7 +255,13 @@ var _ = SIGDescribe("DNS", func() {
 		validateDNSResults(f, pod1, append(wheezyFileNames, jessieFileNames...))
 	})
 
-	It("should provide DNS for ExternalName services", func() {
+	/*
+		Release: v1.15
+		Testname: DNS, for ExternalName Services
+		Description: Create a service with externalName. Pod MUST be able to resolve the address for this service via CNAME. When externalName of this service is changed, Pod MUST resolve to new DNS entry for the service.
+		Change the service type from externalName to ClusterIP, Pod MUST resolve DNS to the service by serving A records.
```
Comment (smarterclayton) [inline test/e2e/network/dns.go]: This is an excellent description!

**Pull request merged**